UM/Application.py

```diff
@@ -348,4 +357,57 @@ def getInstallPrefix():
         else:
             return os.path.abspath(os.path.join(os.path.dirname(sys.executable), ".."))
 
+    def _mrproper(self, only_user = False):
+        path_list = []
+
+        if not only_user:
+            if Platform.isWindows():
+                os.path.join(os.path.expanduser(), )
+                dir_patterns = ["\\AppData\\Local\\%s" %self._application_name,
+                                ]
+                file_patterns = []
+            elif Platform.isOSX():
+                dir_patterns = ["Library/Application Support/%s" %self._application_name,
+                                ]
+                file_patterns = []
+            elif Platform.isLinux():
+                dir_patterns = ['.config/%s' %self._application_name,
+                                '.cache/%s' %self._application_name,
+                                '.local/share/%s' %self._application_name,
+                                ]
+                file_patterns = ['.config/%src' %self._application_name,
+                                 ]
+            else:
+                print("Unknown platform. Nothing to delete!")
+                return
+
+            user_basedir = os.path.split(os.path.expanduser("~"))[0]
+            user_paths = [os.path.join(user_basedir, entry) for entry in os.listdir(user_basedir)]
+            for user_path in user_paths:
+                for dir_pattern in dir_patterns:
+                    path_list.append(os.path.join(user_path, dir_pattern))
+                for file_pattern in file_patterns:
+                    path_list.append(os.path.join(user_path, file_pattern))
+
+        else:
+            if Platform.isLinux() or Platform.isOSX() or Platform.isWindows():
+                path_list += [Resources.getConfigStoragePath(),
+                              Resources.getDataStoragePath(),
+                              Resources.getStoragePathForType(Resources.Cache),
+                              ]
+            else:
+                print("Unknown platform. Nothing to delete!")
+                return
```

Comment:
Please only use `Resources.getConfigStoragePath` and such to get these paths. Also don't forget `Resources.getSearchPaths`. This should automatically work whenever we feel like changing the path locations on a rainy afternoon.

Reply:
I can only imagine it won't be able to catch other users' config directories. If I'm not wrong we are expanding the home variable inside `Resources`, therefore the info is lost.

Reply:
Yes, we are expanding the home variable there. This won't be a problem any more when the feature for removing other users' configurations is removed.